File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int furthest = 0;
        for (int i = 0; i <= furthest && i < nums.size(); ++i) {
            furthest = max(furthest, i + nums[i]);
        } 
        return furthest >= nums.size() - 1;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
from heapq import *
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = [0] * 2001
        for num in nums:
            count[num + 1000] += 1
        ret = []  # initialize heap
        for num, freq in enumerate(count):
            # To make max heap, we use -freq to reverse the min heap
            heappush(ret, (-freq, num - 1000))  
        return [heappop(ret)[1] for _ in range(k)]